tools/reviewer/aa/aa_tool.sh

```diff
@@ -10,16 +10,45 @@
 # To compile aa from a workspace: 'export AA_FORCE_COMPILE_WS=<>'
 # To undo: 'unset AA_FORCE_COMPILE_WS'
 
-# TODO:
-# 1. Add function set_BAZEL_WORKSPACE(), to set variable to either head or debug ws.
-# 2. Add function bazel_run(force_build) to run a bazel command:
-#    1. cd into workspace
-#    2. Check if file exists
-#    3. If not (or if force_build), bazel build it
-#    4. Exit if build fails
-#    5. Run command
-# 3. Use bazel_run() in _aa_completions() and in aa(), set_BAZEL_WORKSPACE() in aa()
-# 4. Make sure that after this, `aa` can be run from outside of a workspace (but inside base).
+function set_STARTUP_OS_REPO() {
+    set_AA_BASE
+    if [[ -z "$AA_BASE" ]]; then
+    echo "BASE file not found in path until root"
+    return 1
+
+    fi
+    if [[ -z "$AA_FORCE_COMPILE_WS" ]]; then
+        # we want to use `aa` from head
+        export STARTUP_OS_REPO="$AA_BASE/head/startup-os/"
+    else
+        # there's a workspace we want to use `aa` from
+        export STARTUP_OS_REPO="$AA_BASE/ws/$AA_FORCE_COMPILE_WS/startup-os/"
+    fi
+}
+
+function bazel_run() {
+    local target=$1
+    shift;
+    local force_compile=$1
+    shift;
+    local args="$@"
+    local binary_for_target=$(echo "$target" | sed -e 's|//||g; s|:|/|g; s|^|bazel-bin/|g')
```

Comment:
Maybe place `s|^|bazel-bin/|g` first, so that the replaces are in order?
Also please explain what the `sed` command does.

Reply:
@oferb
>Maybe place s|^|bazel-bin/|g first, so that the replaces are in order?

we could do that

>Also please explain what the sed command does.

`sed` replaces bazel target name with name of the binary `bazel build` would produce
by doing the following (`sed` commands are split by `;`):
1. replace `//` with nothing
2. replace `:` with `/`
3. add `bazel-bin/` to the beginning